docs/proposals/performance-related-monitoring.md

```diff
@@ -0,0 +1,126 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/performance-related-monitoring.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Performance Monitoring
+
+## Reason for this document
+
+This document serves as a place to gather information about past performance regressions, their reason and impact and discuss ideas to avoid similar regressions in the future.
+Main reason behind doing this is to understand what kind of monitoring needs to be in place to keep Kubernetes fast.
+
+## Known past and present performance issues
+
+### Higher logging level causing scheduler stair stepping
+
+Issue https://github.com/kubernetes/kubernetes/issues/14216 was opened because @spiffxp observed a regression in scheduler performance in 1.1 branch in comparison to `old` 1.0
+cut. In the end it turned out the be caused by `--v=4` (instead of `--v=2`) flag in the scheduler together with the flag `--alsologtostderr` which disables batching of log
+lines. This caused wired behavior of the whole component.
```

Comment:
Two comments:
- while it was initially due to running with `--v=4`, it came up again due to some logging statements that had no V level applied, and couldn't be filtered away even with `--v=1` (hence https://github.com/kubernetes/kubernetes/issues/17449)
- we were running with the default settings (equivalent to `--logtostderr=true`); I notice everyone keeps talking about `--alsologtostderr=true` which is a non-default setting that only applies if you're running with with a non-default setting of `--log-dir=some-nonempty-string`... point being, running with the defaults is what led us here, but everyone seems to be implying they use non-default settings; should the defaults be changed?